[chart] Fix custom Tooltip demos (#15631)
mui/mui-x: docs/data/charts/tooltip/tooltip.md

// With composition
<ChartContainer>
  // ...
  <ChartsTooltipContainer trigger="item">
    <CustomItemTooltipContent />
  </ChartsTooltipContainer>
</ChartContainer>
```

:::warning
Do not skip ChartsTooltipContainer rendering if the tooltip has no data to display.
For example the following code does not work.

```jsx
if (tooltipData === null) {
  return null;
}

return (
  <ChartsTooltipContainer trigger="item">
    {/** My content **/}
  </ChartsTooltipContainer>
);
```

> The ChartsTooltipContainer needs needs to be render before the pointer enters the SVG, because it uses this event to get pointer type.

Author: ```suggestion
The ChartsTooltipContainer must render before the pointer enters the SVG because it uses this event to get the pointer type.
```
